ambari-common/src/main/python/resource_management/libraries/execution_command/execution_command.py

```diff
@@ -54,32 +54,98 @@ def __get_value(self, key, default_value=None):
     except:
       return default_value
 
+  """
+  Global variables section
+  """
+
+  def get_module_configs(self):
+    return self.__module_configs
+
   def get_module_name(self):
     return self.__get_value("serviceName")
 
-  def get_stack_name(self):
-    return self.__get_value("hostLevelParams/stack_name")
+  def get_component_type(self):
+    return self.__get_value("role")
 
-  def get_stack_version(self):
-    return self.__get_value("hostLevelParams/stack_version")
+  def get_component_instance_default_name(self):
+    return self.__get_value("default")
 
-  def get_new_stack_version_for_upgrade(self):
-    """
-    New Cluster Stack Version that is defined during the RESTART of a Rolling Upgrade
-    :return:
-    """
-    return self.__get_value("commandParams/version")
+  def get_servicegroup_name(self):
+    return self.__get_value("serviceGroupName")
 
   def get_host_name(self):
     return self.__get_value("hostname")
 
+  def get_cluster_name(self):
+    return self.__get_value("clusterName")
+
+  """
+  Host related variables section
+  """
+
+  def get_mpack_name(self):
+    return self.__get_value("hostLevelParams/stack_name")
+
+  def get_mpack_version(self):
+    return self.__get_value("hostLevelParams/stack_version")
+
   def get_java_home(self):
     return self.__get_value("hostLevelParams/java_home")
 
   def get_java_version(self):
     java_version = self.__get_value("hostLevelParams/java_version")
     return int(java_version) if java_version else None
 
-  def get_module_configs(self):
-    return self.__module_configs
+  def get_jdk_location(self):
+    java_version = self.__get_value("hostLevelParams/jdk_location")
+    return int(java_version) if java_version else None
+
+  def check_agent_stack_want_retry_on_unavailability(self):
+    return bool('hostLevelParams/agent_stack_retry_on_unavailability')
+
+  def get_agent_stack_retry_count_on_unavailability(self):
+    return int('hostLevelParams/agent_stack_retry_count')
```

Comment:
Should be ambariLevelParams/agent_stack_retry_count. 

Also need to handle default value of None if incase the property is not present in command.json